ghub.el

```diff
@@ -678,7 +678,7 @@ has to provide several values including their password."
   (let* ((user (ghub--ident username package))
          (token
           (or (car (ghub--auth-source-get (list :secret)
-                     :host host :user user))
+                     :host host :user user :require '(:user)))
```

Comment:
I agree that we should do this based on your description of what could happen otherwise, but I do not understand why that could happen.

As I understand it we change

> Give me the entry where host is HOST and user is USER.

to

> Give me the entry where host is HOST and user is USER and user is not undefined.

If we already say that user must be USER, why do we also have to say that user may not be nil? It seems to me that the former implies the latter and I don't see what use-case there could be to simply ignore "user is USER" as `auth-source` apparently does without `:require '(:user)`.

Reply:
> Or maybe I misunderstood completely.

No, it's probably I who have misunderstood. My reasoning behind #60 and this PR was to change what I thought was

    Give me the entry where host is HOST or user is USER.

to

    Give me the entry where host is HOST or user is USER and user is not undefined.

In other words, I thought `auth-source-search` queries are disjunctive, rather than conjunctive. This is why I asked

> Should this also :require the properties :host and :secret, to be super explicit?

The reason I thought this is probably due to some old bugs like [bug#22188](https://debbugs.gnu.org/cgi/bugreport.cgi?bug=22188) and experimental errors of my own years ago, where more entries than I expected were showing up.

I can no longer, however, reproduce a disjunctive query, so if I stumble on one again I will know that it is a bug in another package (such as my personal `auth-source` backend).

I'll close both this PR and #60 for now, but feel free to reopen if you still think my initial report was accurate. Thanks.